lib/solidus_importer/processors/base.rb

```diff
@@ -3,40 +3,33 @@
 module SolidusImporter
   module Processors
     class Base
-      include ::SolidusImporter::Utils
-
-      def initialize(importer, row)
-        @importer = importer
-        @row = row
-        @errors = nil
-        @messages = nil
-        @new_record = false
-      end
-
-      def invalid_keys
-        (@row.data&.keys || []) - (mapping&.keys || [])
-      end
-
       def options
         {}
       end
 
-      def update_context(entity, saved)
-        {
-          id: entity.id,
-          class_name: entity.class.name,
-          new_record: @new_record
-        }.merge(
-          if saved
-            { success: true }
-          else
-            { success: false, messages: @messages || entity.errors.full_messages.join(', ') }
-          end
-        )
+      def prepare_context(entity:, new_record:, success:)
+        if success
+          {
+            success: true,
+            id: entity.id,
+            class_name: entity.class.name,
+            entity: entity,
+            new_record: new_record
+          }
```

Comment:
Actually:
- I have to pass `new_record` because I need to call it before calling `save` method;
- `id` and `class_name` are there because they are useful when the context is serialized (like in log processor) - but I could extract that fields in the specific processors that require a serialization.